R-package/tests/testthat/test_lgb.Booster.R

```diff
@@ -172,15 +172,22 @@ test_that("Loading a Booster from a text file works", {
     data(agaricus.test, package = "lightgbm")
     train <- agaricus.train
     test <- agaricus.test
+    params <- list(
+        num_leaves = 4L
+        , boosting = "rf"
+        , bagging_fraction = 0.8
+        , bagging_freq = 1L
+        , force_col_wise = TRUE
+        , categorical_feature = c(1L, 2L)
+        , interaction_constraints = list(c(1L, 2L), 1L)
+        , learning_rate = 1.0
+        , objective = "binary"
+        , verbosity = VERBOSITY
+    )
```

Comment:
I haven't reviewed this PR too thoroughly yet, but want to make one comment right now about testing as I think it'll help you.

Similar to https://github.com/microsoft/LightGBM/pull/4802#discussion_r786419637.

I believe that both the R package and Python packages should have tests that check at least one parameter per type, to catch type-specific issues.

In `config_auto.cpp`, I see the following types

* `bool`
* `double`
* `int`
* `string`
* `vector<double>`
* `vector<int>`
* `vector<string>`

I think this PR is missing tests for these types:

* `vector<double>`
* `vector<string>`

Can you please modify the R and Python tests to check parameters with those types? Since they are sometimes very specialized (for example, the only `vector<double>` is `cegb_*` parameters), I think it would make sense to add *new* test cases narrowly checking those types.